### DISCUSSION TITLE
how deos the fabric dependency syntax work for ignoring stuff

### DISCUSSION THREAD

**Discussion by fayer3** [Kira-NT/mc-publish#110] [Q&A]
I have a speciffic old sodium version marked as breaking in my mod.json, and mc-publish takes that and marks complete sodium as incomplatible on modrinth.
how can I tell mc-publish to ignore the sodium dependency, I saw there is a syntax for it in the readme, but I'm a bit confused.

my dependency: 
```
  "breaks": {
    "sodium": ["0.4.11+build.25", "0.4.11+build.26"]
  },
```
would that the right thing to add, to ignore it?

```
"custom": {
  "mc-publish": {
    "dependencies": [
      "sodium(incompatible)#(ignore)"
    ]
  }
}
```

**Comment by Kira-NT** [comment]
Turns out you're not that confused after all! :)

Yes, this is the exact solution I would suggest to you myself - it will tell `mc-publish` to ignore the Sodium "dependency" completely.

If you ever find yourself unsure whether the dependency pattern is properly formatted, until I finish the proper documentation on the project with some useful interactive elements for cases like this one, you can manually check it against this regex (for example, on [regex101](https://regex101.com/)):

```regex
^\s*(?<id>[^@{(#]+)(@(?<versionRange>[^@{(#]*))?(?:\((?<type>[^@{(#]*)\))?(?<aliases>(?:\{[^:=]+(?:=|:)[^}]*\})+)?(?<ignore>#\(ignore(?::(?<ignoredPlatforms>[^)]*))?\))?\s*$
```

---

Also, since one of `mc-publish`'s goals is to be as zero-configy as possible, I would love to support cases like yours out of the box, to minimize the hassle on the user's side. Currently, `mc-publish` blindly takes the type of the specified dependency and passes it directly to Modrinth/CurseForge, without any regard to the provided version range of the said dependency. Maybe for incompatible "dependencies", I should add a small heuristic that only passes such entries to the target platform if a user specifies a "*" range, i.e., their mod is not compatible with the declared project at all and under no circumstances?

So,

```json
  "breaks": {
    "sodium": "*"
  },
```

will be treated as "incompatible," in Modrinth terms, just as usual, and

```json
  "breaks": {
    "sodium": ["0.4.11+build.25", "0.4.11+build.26"]
  },
```

will be treated as "conflicting" (and Modrinth doesn't have that, so this entry won't be propagated to Modrinth at all).

How does that sound?

**Reply by fayer3** [reply]
probably fine?
The ignoring is enough for me atm.
